### DISCUSSION TITLE
使用自带证书实现reality的方法（偷自己？）

### DISCUSSION THREAD

**Discussion by guigeng** [XTLS/Xray-core#2129] [General]
1. 服务器部署一个http2网站
 比如`https://nextcloud.xxx.com`
hosts设置`127.0.0.1 nextcloud.xxx.com`

2. xray设置
```json
                    "dest": "nextcloud.xxx.com:443", // 必填，格式同 VLESS fallbacks 的 dest
                    "xver": 0, // 选填，格式同 VLESS fallbacks 的 xver
                    "serverNames": [ // 必填，客户端可用的 serverName 列表，暂不支持 * 通配符
                        "nextcloud.xxx.com"
                    ]
```

3. cdn设置
nextcloud.xxx.com到源服务器

测试结果：
客户端设置sni为nextcloud.xxx.com连接正常
浏览器访问https://nextcloud.xxx.com页面显示正常

看RPRX发过一条channel消息：
```
今天想发个频道消息，那就解释一下 REALITY example 中提到的“目标 IP 冷门或许更好”吧，它的意思是：中间人找不到你的 dest 就更好了。最近 @yuhan6665 写的 RealiTLScanner 突然火了，你可以用它扫描出某个 IP 段的网站，然后你查其域名的 DNS 解析，如果没有指向被扫出的 IP，而是指向了 CDN，very good。

Update：也就是说 GFW 会以为你的 REALITY 就是源站、仅此一家。

```
以上是在K3S环境运行的。

这样的做法是不是跟RPRX所说的是一样的？

**Comment by toyo2333** [comment]
请教

0，这几个月效果用得如何？
1，https://nextcloud.xxx.com 打开的内容是什么？
2，网站都用了443了，那Xray用什么端口呢？原作者说不建议xray用443以外的端口
3，CDN设置这一部分能再详细讲讲么？
